scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -797,10 +807,203 @@ cncliPTsendslots() {
+    echo "ERROR: Koios epoch_stake_snapshot history query failed."
+    return 1
+  fi
+
+  pool_stake_hist=$(jq -r '.[].active_stake' <<< "${pool_hist}")
+  active_stake_hist=$(jq -r '.[].active_stake' <<< "${epoch_hist}")
+
+  echo "Processing previous epoch: ${one_or_many}"
+  stake_param_prev="--active-stake ${active_stake_hist} --pool-stake ${pool_stake_hist}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${one_or_many}" --ledger-set prev ${stake_param_prev} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+
+  return 0
+}
+
+processAllEpochs() {
+  getCurrNextEpoch
+  IFS=' ' read -r -a epochs_array <<< "$EPOCHS"
+
+  for EPOCH in "${epochs_array[@]}"; do
+    if ! getConsensus "${EPOCH}"; then echo "ERROR: Failed to fetch protocol parameters for epoch ${EPOCH}.; return 1"; fi
+     if [[ "$EPOCH" == "$curr_epoch" ]]; then
+       runCurrentEpoch
+     elif [[ "$EPOCH" == "$next_epoch" ]]; then
+       runNextEpoch
+     else
+       runPreviousEpochs
+     fi
+  done
+
+  id=1
+  while IFS= read -r row; do
+    echo "$id,$row" >> "$csvfile"
+    ((id++))
+  done < "$tmpcsv"
+
+  sqlite3 "$BLOCKLOG_DB" <<EOF
+DELETE FROM epochdata;
+VACUUM;
+.mode csv
+.import '$csvfile' epochdata
+REINDEX epochdata;
+EOF
+
+  row_count=$(sqlite3 "$BLOCKLOG_DB" "SELECT COUNT(*) FROM epochdata;")
+  echo "$row_count rows have been loaded into epochdata table in blocklog db"
+  echo "~ CNCLI epochdata table load completed ~"
+
+  rm $csvfile $tmpcsv
+}
+
+processSingleEpoch() {
+ if isNumber "$1"; then
+     getCurrNextEpoch
+     IFS=' ' read -r -a epochs_array <<< "$EPOCHS"
+
+   if [[ ! " ${epochs_array[@]} " =~ "$1" ]]; then
+     echo "No slots found in blocklog table for epoch ${EPOCH}."
+     echo
+     exit 1
+   fi
+
+   if ! getConsensus "${1}"; then echo "ERROR: Failed to fetch protocol parameters for epoch ${1}.; return 1"; fi
+
+   if [[ "$1" == "$curr_epoch" ]]; then
+      runCurrentEpoch
+   elif [[ "$1" == "$next_epoch" ]]; then
+      runNextEpoch
+   else
+      runPreviousEpochs
+   fi
+
+   ID=$(sqlite3 "$BLOCKLOG_DB" "SELECT max(id) + 1 FROM epochdata;")
+   csv_row=$(cat "$tmpcsv")
+   modified_csv_row="${ID},${csv_row}"
+   echo "$modified_csv_row" > "$onerow_csv"
+
+   sqlite3 "$BLOCKLOG_DB" "DELETE FROM epochdata WHERE epoch = '$1';"
+   sqlite3 "$BLOCKLOG_DB" <<EOF
+.mode csv
+.import "$onerow_csv" epochdata
+REINDEX epochdata;
+EOF
+   row_count=$(sqlite3 "$BLOCKLOG_DB" "SELECT COUNT(*) FROM epochdata WHERE epoch = '$1';")
+   echo "$row_count row has been loaded into epochdata table in blocklog db for epoch '$1'"
+   echo "~ CNCLI epochdata table load completed ~"
+   echo
+
+   rm $onerow_csv $tmpcsv
+ else
+   echo "No data found in $tmpcsv."
+   exit 1
+ fi
+}
+
+cncliEpochData() {
+  cncliInit
```

Comment:
remove, cncliInit already called before cncliEpochData is called.

Reply:
removed cncliInit from cncliEpochData